TMU/SL/TMU_Module.c

```diff
@@ -0,0 +1,384 @@
+/*
+ * TMU_Module.c
+ *
+ * Created: 24/02/2020 17:40:28
+ *  Author: FATHI
+ */ 
+
+
+/************************************************************************/
+/*			          	 INCLUDES		                     	        */
+/************************************************************************/
+
+#include "TMU_Module.h"
+#include "softwareDelay.h "
+
+/************************************************************************/
+/*			         static Structures Definitions	                       */
+/************************************************************************/
+
+typedef struct TMU_Task_s
+{
+	TMU_funptr pfun_TMU;
+	uint32_t Time_delay;
+	uint8_t periodicity;
+	uint32_t Counts;
+	uint16_t ID;
+	
+}TMU_Task_s;
+
+
+/************************************************************************/
+/*                GLOBAL STATIC VARIABLES                           */
+/************************************************************************/
+
+static TMU_Task_s  gapstr_Tasks_Buffer[ MAX_NUM_OF_TASKS ] = {{0}};
```

Comment:
Use the init function to init all your variables and use deinit to return those back to the init values